packages/hydrogen/src/product/VariantSelector.ts

```diff
@@ -186,7 +195,11 @@ function useVariantPath(handle: string, productPath: string) {
       ? `${match![0]}${productPath}/${handle}`
       : `/${productPath}/${handle}`;
 
-    const searchParams = new URLSearchParams(search);
+    const searchParams = new URLSearchParams(
+      optimistic && navigation.state === 'loading'
+        ? navigation.location.search
+        : search,
+    );
```

Comment:
Yeah, added. Honestly it might be okay to _always_ look at the destination search params, and not introduce the `optimistic` prop. Technically that's a breaking change, but now that this is apart of 2024.7, we could make that change?

If we make a breaking change the implication would be people that use `VariantSelector` but not the new `useOptimisticProduct` hook would get the following experience:

https://screenshot.click/12-28-okx44-2y7wv.mp4

Not the option immediately changes. So in a sense _everyone_ gets some level of optimistic updates. But it's not totally optimistic. See how it takes a few seconds for the "add to cart" button to change states.

Reply:
Yeah can we soft `@deprecate` it and print something?